### DISCUSSION TITLE
L09A, unknown root password. Current version 1.46.16.

### DISCUSSION THREAD

**Discussion by MyNameIsAlvar** [duhow/xiaoai-patch#47] [Q&A] [answered]
Hi all.

Awesome project you guys have here.

I got my hands on L09A. Current FW is version 1.46.16, app says that update to 1.76.6 is available, but downloading it fails. Maybe it's my problem, who knows. But that is not the issue here.

I was able to attach serial console and into u-boot. Sadly I'm out of ideas how to get/reset root password.

I tried simply:
U-Boot > setenv bootargs "${bootargs} reboot=cold,hard emergency init=/bin/sh" and then nboot boot0, but nothing useful happened.

I attached [boot log](https://github.com/duhow/xiaoai-patch/files/10960463/L09a.txt). Maybe someone can point me to the right direction.

**Comment by duhow** [comment, marked as answer]
Try getting the factory password by using this script:
[mi_passwd.html](https://github.com/duhow/xiaoai-patch/files/6687576/mi_passwd.html.txt)

You can also flash from u-boot using `loady` and writing the content to NAND (didn't test it here, so be careful to modify only the second factory partition!)

**Comment by MyNameIsAlvar** [comment]
Thank you for the link. I had already downloaded it while browsing issues, but i forgot what and where.

mico login: root
Password:


BusyBox v1.27.2 () built-in shell (ash)

  _____  _              __     ___  ___  _____
 |     ||_| ___  ___   |  |   |   || . ||  _  |
 | | | || ||  _|| . |  |  |__ | | ||_  ||     |
 |_|_|_||_||___||___|  |_____||___||___||__|__|
------------------------------------------------

      ROM Type:release / Ver:1.46.16
------------------------------------------------
root@mico:~# df -h
Filesystem                Size      Used Available Use% Mounted on
/dev/mtdblock4           25.8M     25.8M         0 100% /
tmpfs                    57.9M    372.0K     57.6M   1% /tmp
tmpfs                   512.0K         0    512.0K   0% /dev
/dev/ubi0_0              13.4M    672.0K     12.1M   5% /data
/dev/ubi0_0              13.4M    672.0K     12.1M   5% /etc/shadow

I'll keep you guys posted about my success (or failure).

**Reply by MyNameIsAlvar** [reply]
Update: after reading almost all the discussion and also https://bbs.hassbian.com/thread-8754-1-1.html I had some progress:

steps:

```
root@mico:~# scp root@10.69.10.87:/root/xiaoai/release/lx09a/image-230313-2209 /tmp/
root@mico:~# mtd write /tmp/image-230313-2209 system1
root@mico:~# dd if=/dev/mtd2 of=/tmp/mtd3
root@mico:~# mtd write /tmp/mtd3 boot1
root@mico:~# fw_env -s boot_part boot1
root@mico:~# reboot
```

And voila - i have passwordless root over terminal AND ssh with generated password, so it seems that _something_ is working...

But now what? I assume that i got rid of that Chinese thing. But how can I use this speaker? I'm google-home/assistant flavored person.

It may sound as a stupid question, but bear in mind that I got that speaker from my friend who accidentally ordered L09A instead L09G.

**Reply by duhow** [reply]
The project currently does not support any Voice Assistant platform (Google, Alexa, Apple). Those have propietary software and I don't provide any means to install or run those.

By using the patched image with software, you can use the speaker as a "remote" speaker (Airplay, MPD, UPNP, Snapcast).
Any other service may be working as well, but bear in mind that the reduced free space blocks most of Python-based applications.
